### PR TITLE
Feature/android release hookup

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR includes changes to support the use of the Java release pipeline for Android packages. There isn't much here, just a flag that the pipeline needs to differentiate between a pipeline running out of the Java repo to work around an incompatibility that has crept in deeper in the pipeline.

This PR has an equivalent in the Java repo, as well as a change in the internal build tools repo (which would need to be landed first).

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 1]
```diff
@@ -16,4 +16,5 @@ stages:
       parameters:
         DependsOn: Build
         Artifacts: ${{parameters.Artifacts}}
-        ArtifactName: packages
\ No newline at end of file
+        ArtifactName: packages
+        VerifyVersions: 'false'
```
Comment (JimSuplizio) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml]: No quotes around the boolean value **false**

**Comment by JimSuplizio** [comment]
> This PR includes changes to support the use of the Java release pipeline for Android packages. There isn't much here, just a flag that the pipeline needs to differentiate between a pipeline running out of the Java repo to work around an incompatibility that has crept in deeper in the pipeline.
/CC @weshaggard 
@mitchdenny The java pipeline is tailored towards releasing the Java SDK packages and the story is similar for net, js and python. I don't see the java pipeline becoming less tailored. If there are special considerations (or lack of) for Andriod then should it have its own definition?

**Comment by mitchdenny** [comment]
> > This PR includes changes to support the use of the Java release pipeline for Android packages. There isn't much here, just a flag that the pipeline needs to differentiate between a pipeline running out of the Java repo to work around an incompatibility that has crept in deeper in the pipeline.
> > /CC @weshaggard
> > @mitchdenny The java pipeline is tailored towards releasing the Java SDK packages and the story is similar for net, js and python. I don't see the java pipeline becoming less tailored. If there are special considerations (or lack of) for Andriod then should it have its own definition?

When I first created the release templates I avoided any repo specific operations and instead of focused on processing the artifacts going through the pipeline. That is why you don't see many places in the rest of the templates where files are cloned from the "self" repo ... or at least that was originally the case.

For both Java and JavaScript release pipeline I have/had instances where repos other than our main mono-repos used the same release pipelines (qpid-proton-j-extensions for Java, and I think another one for JavaScript).

All that said, I think it might start to make sense to split them at this point in time since the PR update process is going to start resulting in some fairly ugly forking logic in the release templates. Might be able to get reuse at a lower level via using template fragments.

**Pull request merged**